sorting/merge_sort.c

```diff
@@ -2,138 +2,121 @@
  * @file
  * @brief Implementation of [merge
  * sort](https://en.wikipedia.org/wiki/Merge_sort) algorithm
+ * @author Bama Charan Chhandogi(https://www.github.com/BamaCharanChhandogi)
  */
-#include <stdio.h>
-#include <stdlib.h>
+#include <assert.h>  /// For assert
+#include <stdio.h>   /// For I/O operations
+#include <stdlib.h>  /// For memory operations
 
-/**
- * @addtogroup sorting Sorting algorithms
- * @{
- */
-/** Swap two integer variables
- * @param [in,out] a pointer to first variable
- * @param [in,out] b pointer to second variable
- */
-void swap(int *a, int *b)
+// Function to merge two sorted subarrays into one sorted array
+void merge(int arr[], int left, int mid, int right)
 {
-    int t;
-    t = *a;
-    *a = *b;
-    *b = t;
-}
+    int i, j, k;
+    int n1 = mid - left + 1;
+    int n2 = right - mid;
 
-/**
- * @brief Perform merge of segments.
- *
- * @param a array to sort
- * @param l left index for merge
- * @param r right index for merge
- * @param n total number of elements in the array
- */
-void merge(int *a, int l, int r, int n)
-{
-    int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
-    if (b == NULL)
-    {
-        printf("Can't Malloc! Please try again.");
-        exit(EXIT_FAILURE);
-    }
-    int c = l;
-    int p1, p2;
-    p1 = l;
-    p2 = ((l + r) / 2) + 1;
-    while ((p1 < ((l + r) / 2) + 1) && (p2 < r + 1))
+    // Create temporary arrays to store the left and right subarrays
+    int L[n1], R[n2];
```

Comment:
`ISO C++ forbids variable length array` - I think you should use `malloc` here (and of course `free` later).